src/Compatibility/Microsoft.DotNet.ApiSymbolExtensions/Filtering/DocIdSymbolFilter.cs

```diff
@@ -9,9 +9,18 @@ namespace Microsoft.DotNet.ApiSymbolExtensions.Filtering
     /// Implements the logic of filtering out api.
     /// Reads the file with the list of attributes, types, members in DocId format.
     /// </summary>
-    public class DocIdSymbolFilter(string[] docIdsToExcludeFiles) : ISymbolFilter
+    public class DocIdSymbolFilter : ISymbolFilter
     {
-        private readonly HashSet<string> _docIdsToExclude = new(ReadDocIdsAttributes(docIdsToExcludeFiles));
+        private readonly HashSet<string> _docIdsToExclude;
+
+        public static DocIdSymbolFilter CreateFromFiles(params string[] filesWithDocIdsToExclude)
+            => new DocIdSymbolFilter(ReadDocIdsFromFiles(filesWithDocIdsToExclude));
+
+        public static DocIdSymbolFilter CreateFromLists(params string[] docIdsToExclude)
```

Comment:
Please triple slash document the new public members. I'm not sure why this is still not enforced by the build.

Reply:
This might be the reason:

https://github.com/dotnet/sdk/blob/35a8299fcb0cb13da87c32166ecafd219dd9206c/.editorconfig#L268

I don't want to turn it on for the whole repo. Maybe we can do it for the Compatibility project for now?

Reply:
I tried adding the necessary properties to the csproj, tried adding an .editorconfig to the Compatibility folder, but it still passes. There must be something else overriding it. Let's address that separately to this PR. I'll just add the triple slash comments to these new public APIs.

I opened this issue to track the work item: https://github.com/dotnet/sdk/issues/46283